osquery/tables/system/windows/recent_files.cpp

```diff
@@ -0,0 +1,95 @@
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <osquery/filesystem/filesystem.h>
+#include <osquery/logger/logger.h>
+#include <osquery/sql/sql.h>
+#include <osquery/tables/system/system_utils.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/utils/conversions/windows/strings.h>
+#include <osquery/utils/scope_guard.h>
+#include <osquery/utils/system/errno.h>
+
+#include <filesystem>
+#include <shlobj_core.h>
+
+namespace osquery {
+namespace tables {
+
+void genRecentFilesForUser(QueryData& results,
+                           const std::string& user_path_prefix,
+                           const std::int64_t& uid) {
+  SQL sql(
+      "SELECT filename, path, mtime, shortcut_target_path, "
+      "shortcut_target_type FROM file WHERE shortcut_target_type != 'File "
+      "folder' AND path LIKE '" +
+      user_path_prefix +
+      "\\AppData\\Roaming\\Microsoft\\Windows\\Recent\\%.lnk' ORDER BY mtime "
+      "DESC");

```

Comment:
This table now feels like a wrapper around `file` and `users` tables?
Am not against it, this is simpler to use than memorizing all the expected AppData path.

Reply:
Yes! I can't remember off the top of my head, but this is not the first table to do something similar.